src/gd_heif.c

```diff
@@ -0,0 +1,839 @@
+{
+	_gdImageHeifCtx(im, outfile, quality, codec, chroma);
+}
+
+/*
+  Function: gdImageHeifEx
+
+    <gdImageHeifEx> outputs the specified image to the specified file in
+    HEIF format. The file must be open for writing. Under MSDOS and
+    all versions of Windows, it is important to use "wb" as opposed to
+    simply "w" as the mode when opening the file, and under Unix there
+    is no penalty for doing so. <gdImageHeifEx> does not close the file;
+    your code must do so.
+
+	If _quality_ is -1, a reasonable quality value (which should yield a
+	good general quality / size tradeoff for most situations) is used. Otherwise
+	_quality_ should be a value in the range 0-100, higher quality values
+	usually implying both higher quality and larger image sizes or 200, for
+	lossless codec.
+
+  Variants:
+
+    <gdImageHeifCtx> stores the image using a <gdIOCtx> struct.
+
+    <gdImageHeifPtrEx> stores the image to RAM.
+
+    <gdImageAvifEx> AVIF method counterpart.
+
+  Parameters:
+
+    im          - The image to save.
+    outFile     - The FILE pointer to write to.
+    quality     - Codec quality (0-100).
+    codec       - The output coding format.
+    chroma      - The output chroma subsampling format.
+
+  Returns:
+
+    Nothing.
+*/
+BGD_DECLARE(void) gdImageHeifEx (gdImagePtr im, FILE * outFile, int quality, gdHeifCodec codec, gdHeifChroma chroma)
+{
+	gdIOCtx *out = gdNewFileCtx(outFile);
+	if (out == NULL) {
+		return;
+	}
+	_gdImageHeifCtx(im, out, quality, codec, chroma);
+	out->gd_free(out);
+}
+
+/*
+  Function: gdImageHeif
+
+    Variant of <gdImageHeifEx> which uses the default quality (-1), the
+    default codec (GD_HEIF_Codec_HEVC) and the default chroma
+    subsampling (GD_HEIF_CHROMA_444).
+
+  Parameters:
+
+    im      - The image to save
+    outFile - The FILE pointer to write to.
+
+  Returns:
+
+    Nothing.
+*/
+BGD_DECLARE(void) gdImageHeif (gdImagePtr im, FILE * outFile)
+{
+	gdIOCtx *out = gdNewFileCtx(outFile);
+	if (out == NULL) {
+		return;
+	}
+	_gdImageHeifCtx(im, out, -1, GD_HEIF_CODEC_HEVC, GD_HEIF_CHROMA_444);
+	out->gd_free(out);
+}
+
+/*
+  Function: gdImageHeifPtr
+
+    See <gdImageHeifEx>.
+*/
+BGD_DECLARE(void *) gdImageHeifPtr (gdImagePtr im, int *size)
+{
+	void *rv;
+	gdIOCtx *out = gdNewDynamicCtx(2048, NULL);
+	if (out == NULL) {
+		return NULL;
+	}
+	if (_gdImageHeifCtx(im, out, -1, GD_HEIF_CODEC_HEVC, GD_HEIF_CHROMA_444)) {
+		rv = gdDPExtractData(out, size);
+	} else {
+		rv = NULL;
+	}
+	out->gd_free(out);
+
+	return rv;
+}
+
+/*
+  Function: gdImageHeifPtrEx
+
+    See <gdImageHeifEx>.
+*/
+BGD_DECLARE(void *) gdImageHeifPtrEx (gdImagePtr im, int *size, int quality, gdHeifCodec codec, gdHeifChroma chroma)
+{
+	void *rv;
+	gdIOCtx *out = gdNewDynamicCtx(2048, NULL);
+	if (out == NULL) {
+		return NULL;
+	}
+	if (_gdImageHeifCtx(im, out, quality, codec, chroma)) {
+        rv = gdDPExtractData(out, size);
+    } else {
```

Comment:
seems like indentation is incorrect here

github makes it hard to double check whitespace, so make sure to use tabs everywhere